### DISCUSSION TITLE
The intention of provider's rsa_to_text

### DISCUSSION THREAD

**Discussion by Seulhui** [openssl/openssl#21566] [Q&A]
Dear All,

Hello, I'm recently implementing my new provider from an engine for OpenSSLv3 to use my own cryptographic operation and storage for the private key. It makes X509_REQ, and I compared it to the results of 'X509_REQ_print' from the engine and provider. However, it shows some of the differences may be from providers/implementations/encode_decode/encode_key2text.c rsa_to_text(I use the default provider for encode/decode). 

For example,
Engine result :         Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                RSA Public-Key: (2048 bit)

Provider result :          Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                Public-Key: (2048 bit)

I wonder if there is a specific reason that it doesn't include the algorithm name on the provider, even it knows the algorithm?

Regards,
Seulhui Lee.

**Comment by paulidale** [comment]
I'd guess it was either an oversight on the part of the person coding this, or the information isn't readily available to the printing code.

**Comment by paulidale** [comment]
I've flagged this as a regression since there technically is a change of behaviour.  I'm not that concerned about the change and would be happy calling this a feature request.

**Reply by Seulhui** [reply]
Thank you for your answer! I'll wait for the changes about it.
